helsingborg-stad/helsingborg-io-sls-api
# Wip/viva ms cases officers

Sorry, but this pr includes nonrelated commits. Please focus on following files:
services/stream-eventbridge-ms/lambdas/streamProcessor.js
services/stream-eventbridge-ms/serverless.yml
services/viva-ms/lambdas/syncOfficers.js
services/viva-ms/serverless.yml

What?
Lambda keeping applicant administrators in sync with what's stored in VIVA

How?
'Listening' on database INSERT or MODIFY event from EventBridge
Fetch applicant officers data from VADA via mypage endpoint
Convert applicant officers data to comply with case administrators data model
Comparing officers data with administrator data stored in case (DynamoDB)
Update case administrator data if not in sync with VIVA officers

How to test
Have nodejs version 12 installed
run yarn --no-lock-file from repo root
cd services/stream-eventbridge-ms
run sls deploy
cd ..
cd viva-ms
run sls deploy

Do PUT or POST request on case api to fire off event

Verify that details on case is updated with administrators in DynamoDB

Tested
- [x] My personal AWS environment
